unitTests/curvatureResistance/curvatureResistance.py

```diff
@@ -0,0 +1,318 @@
+
+        except Exception as e:
+
+            raise unittest.SkipTest("Impossible to create track object. Error was: ", e)
+
+        cls.trackConstantCurvature = copy.deepcopy(cls.trackNoCurvature)
+
+        cls.trackConstantCurvature.importCurvatureTuples(tuples=[[0.0, str(1/cls.constantK), str(1/cls.constantK)]])
+
+
+    def specificCurvatureResistanceForce(self, g, rho):
+        '''
+        Computes specific curvature resistance force [m/s^2]
+
+        - param g: constant of gravity [m/s^2]
+        - param rho: dimensionless rotating mass factor
+        '''
+
+        return g*0.5*abs(self.constantK)/((1-30*abs(self.constantK))*rho)*(abs(self.constantK)<=1/300) + \
+                g*0.65*abs(self.constantK)/((1-55*abs(self.constantK))*rho)*(abs(self.constantK)>1/300)
+
+
+    def solveOptimalControlProblem(self, track, energyOptimal, lossFunction,
+                                   terminalTime, train=None, finalPosition=None, realLossFunction=None):
+        '''
+        Solves a minimum time or minimium energy optimal control problem.
+
+        - param pathToTrackData: absolute path to the json file containing the track dataset
+        - param trackDataFilename: name of json file without extension
+        - param energyOptimal: bool variable. If false the minimum time problem is solved
+        - param lossFunction: a function taking two inputs (f,v) and returning the electrical
+        losses of the train. Used for solving the minimum energy problem.
+        - param terminalTime: time at which the trip must complete. For the minimum time problem this is just an upperbound of the problem.
+        - param finalPosition: position where the trip must end.
+        - param realLossFunction: if provided this loss function is used to compute the real losses of the train. Note that it is not used to
+        solve the minimum energy problem.
+        - return: a dataframe containing the solution of the problem.
+        '''
+
+        v0 = self.speedAtStart
+        vN = self.speedAtEnd
+
+        if (train == None):
+
+            train = Train(train='Intercity')
+
+            train.forceMinPn = 0
+
+        if (finalPosition == None):
+
+            finalPosition = self.finalPosition
+
+        track.updateLimits(positionEnd=finalPosition)
+
+        solverOpts = { "maxIterations": 500,
+                    "numIntervals": 300,
+                    "integrationMethod": "RK",
+                    "integrationOptions": {"order": 4, "numSteps": 1, "numApproxSteps": 1 },
+                    "energyOptimal": energyOptimal,
+                    "minimumVelocity": min(v0, vN)}
+
+        # # # solve problem with perfect efficiency
+        train.powerLosses = lossFunction
+
+        ocp0 = casadiSolver(train, track, solverOpts)
+
+        df0, _ = ocp0.solve(terminalTime, terminalVelocity=vN, initialVelocity=v0)
+
+        if (realLossFunction != None):
```

Comment:
I would not re calculate based on "real" losses, simply check that tests pass on the loss function that the OCP knows